frontend/src/components/main-page/graph-page/DepthSlider.tsx

```diff
@@ -0,0 +1,73 @@
+import { Box, Slider } from '@mui/material';
+import React from 'react';
+
+interface Props {
+  maxDepth: number;
+  onChange: (newValue: number) => void;
+}
+
+export const DepthSlider: React.FC<Props> = (props) => {
+  const controller = useController(props);
+
+  const graphDepthMarks = [
+    {
+      value: props.maxDepth,
+      label: `full depth`,
+    },
+  ];
+  let i = 1;
+  for (; i < props.maxDepth; ++i) {
```

Comment:
I would move this into the Controller. Also I find it a bit unusual to see the iterator variable outside the loop. So I would change it to

``` ts
for (let i = 1; i < props.maxDepth; i++) {
  // ...
}
```

And maybe move the "full depth" one after the loop so that it becomes more natural to read: First we add the labels for 1, 2, 3, ..., and then the "full depth" one.